### DISCUSSION TITLE
ts-node in production

### DISCUSSION THREAD

**Discussion by rolanday** [vikejs/vike#1087] [Discussion]
Hello, I installed react-ts template using `pnpm create vite-plugin-ssr` and see that project is setup to use ts-node in production. `tsconfig.json` has `transpileOnly:  true` so I suppose OK, but I've never run ts-node in production before and makes me uneasy to do so. Is there a config to transpile as part of build? I did so manually and things seem to work, but I'm new to VPS and want to avoid any false starts.

Thanks!

```
  "scripts": {
    "dev": "npm run server:dev",
    "prod": "npm run lint && npm run build && npm run server:prod",
    "build": "vite build",
    "server": "node --loader ts-node/esm ./server/index.ts",
    "server:dev": "npm run server",
    "server:prod": "cross-env NODE_ENV=production npm run server",
    "lint": "eslint . --max-warnings 0"
  },
```

**Comment by brillout** [comment]
Yes, in principle, you shouldn't load TypeScript in production. There could be practical issues about it (e.g. about high mem usage) but, so far, I ain't aware of any.

As long as you don't have issues with it, I'd argue it's fine to use it in production.

If you have a memory constrained deploy env, then transpiling the server before could be worth it.

FYI the goal is to eventually ditch `ts-node` with https://github.com/brillout/vite-plugin-ssr/issues/562.

Btw. would your company be up for [sponsoring](https://github.com/sponsors/brillout)? I'm happy to provide further assistance (with a quicker turnaround).
